.github/workflows/ci.yaml

```diff
@@ -163,14 +163,40 @@ jobs:
       - name: Install Numaplane
         env:
           GOPATH: /home/runner/go
-        run:
-          KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} make start
+        run: |
+          KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} make start-e2e
+
+      - name: Wait for Numaplane to be ready
+        run: |
+          echo 'Waiting for the controller-manager to be ready...'
+          kubectl wait --for=condition=available --timeout=60s deployment/numaplane-controller-manager -n numaplane-system
+
+      - name: Start numaplane process
+        run: |
+          POD_NAME=$(kubectl get pods -n numaplane-system -l app.kubernetes.io/name=controller-manager -o jsonpath="{.items[0].metadata.name}")
+          nohup kubectl exec -n numaplane-system $POD_NAME -- /bin/bash -c '/manager --health-probe-bind-address=:8081 --metrics-bind-address=:8080 --leader-elect' > /dev/null 2>&1 &
+          sleep 20s
 
       - name: Run e2e tests
         env:
           GOPATH: /home/runner/go
         run: KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} ENABLE_POD_LOGS=true make test-${{matrix.case}}-e2e
 
+      - name: Stop numaplane process
+        run: |
+          sleep 60s
+          POD_NAME=$(kubectl get pods -n numaplane-system -l app.kubernetes.io/name=controller-manager -o jsonpath="{.items[0].metadata.name}")
+          kubectl exec -n numaplane-system $POD_NAME -- /bin/bash -c 'pkill -f "/manager --health-probe-bind-address=:8081 --metrics-bind-address=:8080 --leader-elect"'
+          sleep 5s
+          kubectl exec -n numaplane-system $POD_NAME -- /bin/bash -c 'pkill -f "/manager --health-probe-bind-address=:8081 --metrics-bind-address=:8080 --leader-elect"'
+          sleep 30s
+
+      - name: Copy report from Numaplane pod
+        run: |
+          mkdir -p tests/e2e/output/resources/coverage/
+          POD_NAME=$(kubectl get pods -n numaplane-system -l app.kubernetes.io/name=controller-manager -o jsonpath="{.items[0].metadata.name}")
+          kubectl cp $POD_NAME:/coverage -n numaplane-system tests/e2e/output/resources/coverage/
```

Comment:
this is great. So, we can have one or more steps to:
1. aggregate all of the e2e coverage, post that as an artifact
2. post the unit test code coverage as an artifact
3. aggregate e2e and unit test code coverage together and post that as an artifact

Ideal thing is to be able to have html output for these artifacts that we could view in a browser

Reply:
actually, you can see if we need artifacts at all, or if posting to CodeCov gives us everything we need, now that I think about it